packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 - http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import * as Classes from "../../common/classes";
+import * as Keys from "../../common/keys";
+import { removeNonHTMLProps } from "../../common/props";
+import { HTMLInputProps, IControlledProps, IIntentProps, IProps } from "../../common/props";
+
+import { Button } from "../button/buttons";
+import { InputGroup } from "./inputGroup";
+
+export type ButtonPosition = "none" | "left" | "right" | "split";
+
+export interface INumericStepperProps extends IControlledProps, IIntentProps, IProps {
+
+    /**
+     * The button configuration with respect to the input field.
+     * @default right
+     */
+    buttonPosition?: ButtonPosition;
+
+    /**
+     * Whether the input is non-interactive.
+     * @default false
+     */
+    disabled?: boolean;
+
+    /** Ref handler that receives HTML `<input>` element backing this component. */
+    inputRef?: (ref: HTMLInputElement) => any;
+
+    /** Name of icon (the part after `pt-icon-`) to render on left side of input. */
+    leftIconName?: string;
+
+    /** Placeholder text in the absence of any value. */
+    placeholder?: string;
+
+    /**
+     * Increment between successive values when "Shift" is held.
+     * @default 10
+     */
+    majorStepSize?: number;
+
+    /** Maximum value of the input. */
+    max?: number;
+
+    /** Minimum value of the input. */
+    min?: number;
+
+    /**
+     * Increment between successive values when "Alt" is held.
+     * @default 0.1
+     */
+    minorStepSize?: number;
+
+    /**
+     * Increment between successive values when no modifier keys are held.
+     * @default 1
+     */
+    stepSize?: number;
+
+    /**
+     * Value to display in the input field
+     * @default ""
+     */
+    value?: string;
+
+}
+
+export interface INumericStepperState {
+    shouldSelectAfterUpdate: boolean;
+    value: string;
```

Comment:
ok i'm pretty sure you've implemented _uncontrolled_ usage here, where the component maintains its own state. we'll want to support controlled usage (even at the expense of uncontrolled usage) as it's practically essential for form components. we can talk this through tomorrow.